pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -1141,7 +1178,7 @@ func (s *sqlEntityServer) List(ctx context.Context, r *entity.EntityListRequest)
 		// if we are looking for deleted entities, we need to use the labels column
 		if r.Deleted {
 			for labelKey, labelValue := range r.Labels {
-				entityQuery.addWhere(s.dialect.Quote("labels")+" LIKE ?", "%\""+labelKey+"\":\""+labelValue+"\"%")
+				entityQuery.addWhere(s.dialect.Quote("labels")+" LIKE ?", `%"`+labelKey+`":"`+labelValue+`"%`)
```

Comment:
Define the following at the package level:
```golang
const sqlLikeEscape = "#" // any ASCII ok

var sqlLikeEscapeReplacer = strings.NewReplacer("%", sqlLikeEscape+"%", "_", sqlLikeEscape+"_")

func escapeJSONStringSQLLike(s string) (string, error) {
	var b strings.Builder
	// we forcibly need to JSON-encode since it escapes some characters on its own, otherwise we wouldn't match sometimes
	if err := json.NewEncoder(&b).Encode(s); err != nil{
		return "", err
	}
	return sqlLikeEscapeReplacer.Replace(b.String()), nil
}
```
Then fix the `LIKE` so we have properly sanitized strings
```suggestion
						// TODO: handle errors
												labelKeySQL, err := escapeJSONStringSQLLike(labelKey)
						labelValSQL, err := escapeJSONStringSQLLike(labelKey)
																entityQuery.addWhere(s.dialect.Quote("labels")+" LIKE ? ESCAPE %", `%`+labelKeySQL+`:`+labelValSQL+`"%`, sqlLikeEscape)
```